src/App.tsx

```diff
@@ -1,24 +1,166 @@
+      ...newTodoToFetch,
+      id: 0,
+    };
+
+    setIsTodoAdding(true);
+    setTempTodo(newTodoToShow);
+
+    try {
+      const addedTodo = await addTodo(newTodoToFetch);
+
+      setTodos(prevTodos => [...prevTodos, addedTodo]);
+    } catch (error) {
+      setErrorType('add');
+    } finally {
+      setIsTodoAdding(false);
+      setTempTodo(null);
+    }
+  }, [title]);
+
+  const deleteTodoFromServer = useCallback(async (todoId: number) => {
+    setRemovingTodoIds(prevTodoIds => [...prevTodoIds, todoId]);
+
+    try {
+      await deleteTodo(todoId);
+
+      const updatedTodos = todos.filter(todo => todo.id !== todoId);
+
+      setTodos(updatedTodos);
+    } catch (error) {
+      setErrorType('delete');
+    } finally {
+      setRemovingTodoIds(prevTodoIds => prevTodoIds
+        .filter((id) => id === todoId));
+    }
+  }, [todos]);
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const clearCompletedTodos = async () => {
+    const completedTodosIds = [...todos]
+      .filter(todo => todo.completed)
+      .map((todo) => todo.id);
+
+    setRemovingTodoIds(completedTodosIds);
+
+    Promise.all(completedTodosIds.map((id) => deleteTodo(id)))
+      .then(() => {
+        const activeTodos = todos
+          .filter(todo => !todo.completed);
+
+        setTodos(activeTodos);
+      })
+      .catch(() => {
+        setErrorType('delete');
+      })
+      .finally(() => setRemovingTodoIds(prevTodoIds => prevTodoIds
+        .filter((id) => !completedTodosIds.includes(id))));
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          title={title}
+          setTitle={setTitle}
+          onAdd={addTodoToServer}
+          hasActive={hasActive}
+          setErrorType={setErrorType}
+          isTodoAdding={isTodoAdding}
+        />
+        {(!!todos.length || tempTodo) && (
+          <>
+            <TodoList
+              tempTodo={tempTodo}
+              todos={filteredTodos}
+              removingTodoIds={removingTodoIds}
+              deleteTodoFromServer={deleteTodoFromServer}
+            />
+
+            <Footer
+              todos={todos}
+              selectedFilter={selectedFilter}
+              onChange={setSelectedFilter}
+              clearCompletedTodos={clearCompletedTodos}
+            />
+          </>
+        )}
+      </div>
+
+      {errorType ? (
+        <ErrorNotification
+          errorType={errorType}
+        />
+      ) : null}
```

Comment:
use [conditional rendering](https://react.dev/learn/conditional-rendering) instead of ternary